src/twisted/internet/base.py

```diff
@@ -1030,7 +1030,9 @@ def runUntilCurrent(self) -> None:
         threadpoolShutdownID = None
 
         def _initThreads(self) -> None:
-            self.installNameResolver(_GAIResolver(self, self.getThreadPool))
+            self.installNameResolver(
+                _GAIResolver(cast(IReactorThreads, self), self.getThreadPool)
+            )
```

Comment:
I think this is safe -- if you have threads than your reactor likely implements `IReactorThreads`? The underlying code assumes it anyway.

Alternately I can add an assert:

```python
assert isinstance(self, IReactorThreads)
```

Reply:
That's not how you do that assertion, it would be something more like `asThreadReactor = IReactorThreads(self)`.

(But also let's not lose *too* much sleep over this, we do not test in an un-threaded configuration and I somewhat doubt that it works any more).